core/templates/dev/head/pages/exploration_editor/statistics_tab/LearnerActionRenderServiceSpec.js

```diff
@@ -0,0 +1,329 @@
+        this.ps.recordExplorationQuitAction('stateName2', 120);
+
+        var learnerActions = this.ps.getPlaythrough().actions;
+        var displayBlocks = this.lars.getDisplayBlocks(learnerActions);
+
+        expect(displayBlocks).toEqual([[
+          this.laof.createNew(
+            'ExplorationStart', {
+              state_name: {
+                value: 'stateName1'
+              }
+            }, 1
+          ),
+          jasmine.any(Object),
+          jasmine.any(Object),
+          this.laof.createNew(
+            'ExplorationQuit', {
+              state_name: {
+                value: 'stateName2'
+              },
+              time_spent_state_in_msecs: {
+                value: 120
+              }
+            }, 1
+          )
+        ]]);
+      });
+
+    it('should split up many learner actions into different display blocks.',
+      function() {
+        this.ps.recordExplorationStartAction('stateName1');
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName2', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName2', 'stateName3', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName3', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName2', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName2', 'stateName3', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName3', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName2', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName2', 'stateName3', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName3', 'stateName1', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordExplorationQuitAction('stateName1', 120);
+
+        var learnerActions = this.ps.getPlaythrough().actions;
+        var displayBlocks = this.lars.getDisplayBlocks(learnerActions);
+
+        expect(displayBlocks).toEqual([
+          [
+            this.laof.createNew(
+              'AnswerSubmit', {
+                state_name: {
+                  value: 'stateName1'
+                },
+                dest_state_name: {
+                  value: 'stateName2'
+                },
+                interaction_id: {
+                  value: 'TextInput'
+                },
+                submitted_answer: {
+                  value: 'Hello'
+                },
+                feedback: {
+                  value: 'Try again'
+                },
+                time_spent_state_in_msecs: {
+                  value: 30
+                }
+              }, 1
+            ),
+            jasmine.any(Object),
```

Comment:
Add short comment explaining why it's okay to check for any object here (and elsewhere).